### PR TITLE
ceph code from kernel 4.19.295

### PR DESCRIPTION
**Pull request by benapetr**
I keep encountering weird issues and bugs with CEPH RBD using current 4.19.19 that seems to contain original code from 4.19.19 from many years ago. This patch should contain a huge number of fixes made to ceph code over the years.

### CODE REVIEW

**Comment by stormi** [comment]
Reference forum discussion: https://xcp-ng.org/forum/topic/7789/what-is-kernel-alt

**Comment by stormi** [comment]
Hi. Sorry for the long delay. Only now did I reach the bottom of my todo list. We'll review this and consider it for XCP-ng 8.3.

**Comment by tescande** [comment]
Hi @benapetr,

Thank you for this PR.

Could you rebase your patch on top of master (it has changed since you submitted the PR) and add your Signed-off-by tag to the commit please ?

Otherwise, looks good to me.

**Review by tescande — CHANGES_REQUESTED** [review]
Hi @benapetr,

Thank you for this PR.

Could you rebase your patch on top of master (it has changed since you submitted the PR) and add your Signed-off-by tag to the commit please ?

Otherwise, looks good to me.

File: SPECS/kernel.spec
Comment (tescande) [inline SPECS/kernel.spec:1]: You need to bump the Release version: `Release: %{?xsrel}.2%{?dist}`

And also add a changelog entry.

**Comment by benapetr** [comment]
Hello,

I rebased and signed off my changes, but I don't know which version should be "bumped" and what do you want to see in the changelog

**Comment by benapetr** [comment]
I added what I think might be the correct "version bump" as well as the comment, but I don't know if I incremented correct version, nor if the comment is in correct format

**Comment by benapetr** [comment]
it's rebased again according to your requests

File: SPECS/kernel.spec [hunk 1]
```diff
@@ -668,6 +668,7 @@ Patch582: 0011-Documentation-auxiliary_bus-Clarify-the-release-of-d.patch
 Patch583: 0012-Documentation-auxiliary_bus-Move-the-text-into-the-c.patch
 Patch584: 0013-CP-41018-Make-CONFIG_AUXILIARY_BUS-y-work.patch
 Patch585: abi-version.patch
+Patch1000: ceph.patch
```
Comment (stormi) [inline SPECS/kernel.spec:671]:
```suggestion

# XCP-ng patches
Patch1000: ceph.patch

```

File: SPECS/kernel.spec [hunk 2]
```diff
@@ -996,12 +997,13 @@ fi
 %{?_cov_results_package}
 
 %changelog
+* Thu Jun 06 2024 Petr Bena <benapetr@gmail.com> - 4.19.19-8.0.33.2
+- Integrated ceph code from kernel 4.19.295
```
Comment (stormi) [inline SPECS/kernel.spec:1001]:
```suggestion
* Thu Jun 06 2024 Petr Bena <benapetr@gmail.com> - 4.19.19-8.0.33.2
- Integrated ceph code from kernel 4.19.295

```

Changelog entries are separated by a newline.

**Review by tescande — APPROVED** [review]
Looks good to me. Thanks a lot

**Review by stormi — APPROVED** [review]

**Comment by stormi** [comment]
Koji build towards v8.3-incoming: http://koji.xcp-ng.org/taskinfo?taskID=55826

**Pull request merged**